### PR TITLE
[#435] Calculate correct bounding box when dragging elements

### PR DESCRIPTION
**Pull request by k88hudson**
Still need to figure out how to take rotation into account

### CODE REVIEW

File: src/components/basic-element/basic-element.jsx [hunk 1]
```diff
@@ -226,11 +223,15 @@ var BasicElement = React.createClass({
    * if they'd otherwise run off the page.
    */
   handleTranslation: function(x, y) {
-    var dl = this.dragLimits,
-        w_edge = dl.width,
-        h_edge = dl.height;
+    var rect_width = this.rect.width * this.state.scale;
```
Comment (k88hudson) [inline src/components/basic-element/basic-element.jsx]: When finding the "actual"/visual width and height of the element after transforms, I'm not sure of the formula to take rotation into account -- @Pomax can you maybe help with this?

Comment (k88hudson) [inline src/components/basic-element/basic-element.jsx]: I tried 

```
var rect_width = h*Math.abs(Math.sin(a)) + w*Math.abs(Math.cos(a));
var rect_height = h*Math.abs(Math.cos(a)) + w*Math.abs(Math.sin(a));
```

but that doesn't seem to be working

**Comment by k88hudson** [comment]
@gvn R?

File: src/components/basic-element/basic-element.jsx [hunk 2]
```diff
@@ -226,11 +223,17 @@ var BasicElement = React.createClass({
    * if they'd otherwise run off the page.
    */
   handleTranslation: function(x, y) {
-    var dl = this.dragLimits,
-        w_edge = dl.width,
-        h_edge = dl.height;
+
+    var a = this.state.angle;
```
Comment (gvn) [inline src/components/basic-element/basic-element.jsx]: `a` is pretty non-semantic. Could we just do `angle` or something more descriptive?

File: src/components/basic-element/basic-element.jsx [hunk 3]
```diff
@@ -226,11 +223,17 @@ var BasicElement = React.createClass({
...
+    var rect_width = this.rect.width;
+    var rect_height = this.rect.height;
+
+    var w_edge = PAGE_WIDTH/2 + rect_width/2 - DRAG_BOUNDS_THRESHOLD;
+    var h_edge = PAGE_HEIGHT/2 + rect_height/2 - DRAG_BOUNDS_THRESHOLD;
```
Comment (gvn) [inline src/components/basic-element/basic-element.jsx]: A bit of a nit, but these vars should probably be camelCase.

File: src/components/basic-element/basic-element.jsx (hunk 2 above)
Comment (gvn) [inline src/components/basic-element/basic-element.jsx]: Actually...looks like it's not even being used.

Comment (k88hudson) [inline src/components/basic-element/basic-element.jsx]: oops, you're right, i took that part out

File: src/components/basic-element/basic-element.jsx [hunk 4]
```diff
@@ -226,11 +223,17 @@ var BasicElement = React.createClass({
    * if they'd otherwise run off the page.
    */
   handleTranslation: function(x, y) {
-    var dl = this.dragLimits,
-        w_edge = dl.width,
-        h_edge = dl.height;
+
+    var a = this.state.angle;
+    var rect_width = this.rect.width;
+    var rect_height = this.rect.height;
```
Comment (gvn) [inline src/components/basic-element/basic-element.jsx]: These vars are only being used once each, so probably don't need to be vars.

**Comment by gvn** [comment]
R+ w. nits. Also, would be good to file a follow-on for rotation.

**Pull request merged**